src/Core/Algorithms/BrainStimulator/BiotSavartSolverAlgorithm.cc

```diff
@@ -740,94 +642,72 @@ bool BiotSavartSolverAlgorithm::run(FieldHandle mesh, FieldHandle coil, MatrixHa
       return (false);
     }
   }
-  else if(coil->vmesh()->is_pointcloudmesh())
+  else if (coil->vmesh()->is_pointcloudmesh())
   {
-   if((coil->vfield()->is_lineardata() || coil->vfield()->is_constantdata() ) && coil->vfield()->is_vector())
-   {
-    auto dp = std::unique_ptr<KernelBase>(new DipolesKernel(this, outtype));
-    if( !dp->Integrate(mesh,coil,outdata) )
-      {
-       error("Aborted during integration");
-       return (false);
-      }
-   }
-   else
-   {
-    error("Pointcloud expected with linear vector data.");
-    return (false);
-   }
-  }
-  else if( coil->vmesh()->is_volume() )
-  {
-   if(  coil->vfield()->is_constantdata() && coil->vfield()->is_vector() )
-   {
-   auto vp = std::unique_ptr<KernelBase>(new VolumetricKernel(this, outtype));
-   if( !vp->Integrate(mesh,coil,outdata) )
-      {
-       error("Aborted during integration");
-       return (false);
-      }
-   }
-   else
-   {
-    error("Volumetric mesh expected with constant vector data.");
-    return (false);
-   }
-  }
-  else
+		if ((coil->vfield()->is_lineardata() || coil->vfield()->is_constantdata() ) && coil->vfield()->is_vector())
+		{
+			DipolesKernel dp(this, outtype);
+			if (!dp.integrate(mesh,coil,outdata))
+			{
+				error("Aborted during integration");
+				return (false);
+			}
+		}
+		else
+		{
+			error("Pointcloud expected with linear vector data.");
+			return (false);
+		}
+	}
+  else if (coil->vmesh()->is_volume())
   {
-   error("Unsupported mesh type! Only curve or volumetric.");
-   return (false);
-  }
+		if (coil->vfield()->is_constantdata() && coil->vfield()->is_vector())
+		{
+			VolumetricKernel vp(this, outtype);
+			if (!vp.integrate(mesh,coil,outdata))
+			{
+				error("Aborted during integration");
+				return (false);
+			}
+		}
+		else
+		{
+			error("Volumetric mesh expected with constant vector data.");
+			return (false);
+		}
+	}
+	else
+	{
+		error("Unsupported mesh type! Only curve or volumetric.");
+		return (false);
+	}
 
   return (true);
 }
 
 AlgorithmOutput BiotSavartSolverAlgorithm::run(const AlgorithmInput& input) const
 {
- AlgorithmOutput output;
-
- auto mesh = input.get<Field>(Parameters::Mesh);
- auto coil = input.get<Field>(Parameters::Coil);
-
- MatrixHandle outdata1,outdata2;
+	AlgorithmOutput output;
 
- auto oports = get(Parameters::OutType).toInt();
+	auto mesh = input.get<Field>(Parameters::Mesh);
+	auto coil = input.get<Field>(Parameters::Coil);
 
- if (oports==3) //both are output ports
- {
-  if(!run(mesh, coil, outdata1, 1))
-  {
-    error("Error: Algorithm of BiotSavartSolver failed.");
-  }
-  output[Parameters::VectorBField] = outdata1;
+	auto oports = get(Parameters::OutType).toInt();
+	std::map<int, DenseMatrixHandle> cases;
+	if (oports == 3)
+		cases = {{1, nullptr}, {2, nullptr}};
+	else
+		cases[oports] = nullptr;
 
-  if(!run(mesh, coil, outdata2, 2))
+  for (auto& c : cases)
```

Comment:
I guess this is pedantic of me, but is there a reason c shouldn't be constant?

Reply:
```suggestion
  for (const auto& c : cases)
```

Reply:
Done!